src/core/compression/algorithm.c

```diff
@@ -141,7 +141,7 @@ grpc_compression_algorithm grpc_compression_algorithm_for_level(
     default:
       break;
   }
-  GPR_UNREACHABLE_CODE(return GRPC_COMPRESS_NONE);
```

Comment:
Suspect there should be at least a diagnostic output (gpr_log(GPR_ERROR, ...)) here (although the abort() would be nice)